crates/rlp/src/header.rs

```diff
@@ -11,6 +12,112 @@ pub struct Header {
 }
 
 impl Header {
+    /// Decodes an RLP header from the given buffer.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the buffer is too short or the header is invalid.
+    #[inline]
+    pub fn decode(buf: &mut &[u8]) -> Result<Self> {
+        let payload_length;
+        let mut list = false;
+        match get_next_byte(buf)? {
+            0..=0x7F => payload_length = 1,
+
+            b @ EMPTY_STRING_CODE..=0xB7 => {
+                buf.advance(1);
+                payload_length = (b - EMPTY_STRING_CODE) as usize;
+                if payload_length == 1 && get_next_byte(buf)? < EMPTY_STRING_CODE {
+                    return Err(Error::NonCanonicalSingleByte)
+                }
+            }
+
+            b @ (0xB8..=0xBF | 0xF8..=0xFF) => {
```

Comment:
replace magic numbers with constants?

Reply:
would need separate constants for the - 1 difference because `..` ranges are not allowed, at that point i don't think it's worth it